### PR TITLE
converters old AFM <-> new VisObj

### PR DESCRIPTION
**Pull request by ZdenekVidensky**

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-data-layer-opensource-repository-check/24/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-data-layer-unit-tests-zuul-el7/375/ : FAILURE in 39s
- https://ci.intgdc.com/job/gooddata-data-layer-unit-tests-zuul-docker-el7/19/ : FAILURE in 40s (non-voting)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-data-layer-opensource-repository-check/25/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-data-layer-unit-tests-zuul-el7/376/ : FAILURE in 22s
- https://ci.intgdc.com/job/gooddata-data-layer-unit-tests-zuul-docker-el7/20/ : FAILURE in 23s (non-voting)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-data-layer-opensource-repository-check/26/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-data-layer-unit-tests-zuul-el7/377/ : FAILURE in 45s
- https://ci.intgdc.com/job/gooddata-data-layer-unit-tests-zuul-docker-el7/21/ : FAILURE in 40s (non-voting)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-data-layer-opensource-repository-check/27/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-data-layer-unit-tests-zuul-el7/378/ : SUCCESS in 1m 04s
- https://ci.intgdc.com/job/gooddata-data-layer-unit-tests-zuul-docker-el7/22/ : SUCCESS in 1m 04s (non-voting)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-data-layer-opensource-repository-check/28/ : SUCCESS in 33s
- https://ci.intgdc.com/job/gooddata-data-layer-unit-tests-zuul-el7/379/ : SUCCESS in 3m 56s
- https://ci.intgdc.com/job/gooddata-data-layer-unit-tests-zuul-docker-el7/23/ : SUCCESS in 1m 00s (non-voting)

File: src/legacy/model/VisualizationObject.ts [hunk 1]
```diff
@@ -142,3 +236,34 @@ export interface IVisualizationMetadataResult {
...
+
+export function isPositiveAttributeFilter(filter: AttributeFilter): filter is IPositiveAttributeFilter {
+    return (filter as IPositiveAttributeFilter).positiveAttributeFilter !== undefined;
+}
+
+export function isAbsoluteDateFilter(filter: DateFilter): filter is IAbsoluteDateFilter {
+    return (filter as IAbsoluteDateFilter).absoluteDateFilter !== undefined;
+}
+
+export function isAttribute(bucketItem: BucketItem): bucketItem is IVisualizationAttribute {
```
Comment (ivan-nejezchleb) [inline src/legacy/model/VisualizationObject.ts]: what is a difference between this function and 'isVisualizationAttribute' ?

File: src/legacy/toAFM.ts [hunk 2]
```diff
@@ -262,27 +342,15 @@ function convertTransformation(
...
-    return (uri: string) => {
-        return attributesMap[uri];
-    };
 }
 
 export function toAFM(
     visObj: VisObj.IVisualizationObjectContent,
-    attributesMap: VisObj.IAttributesMap,
+    attributeTypeMap?: AttributeTypeMap,
     measuresMap?: VisObj.IMeasuresMap
```
Comment (ivan-nejezchleb) [inline src/legacy/toAFM.ts]: not used param - remove it

**Pull request closed without merging**